src/libmongoc/doc/mongoc_client_pool_new.rst

```diff
@@ -21,5 +21,13 @@ Parameters
 Returns
 -------
 
-A newly allocated :symbol:`mongoc_client_pool_t` that should be freed with :symbol:`mongoc_client_pool_destroy()` when no longer in use.
+A newly allocated :symbol:`mongoc_client_pool_t` that should be freed with :symbol:`mongoc_client_pool_destroy()` when no longer in use. On error, ``NULL`` is returned and an error may be logged.
```

Comment:
`mongoc_client_new` and `mongoc_client_new_from_uri` both returned `NULL` on error; however, `mongoc_client_pool_new` always returned a pool. I suppose we can't start returning `NULL` on error here since that would introduce a BC break.

Hopefully we _can_ do so for `mongoc_client_pool_new_with_error` since that'd be a new method. Let me know your thoughts.


Returning `NULL` for This may need to be changed

Reply:
I do not think that is the case. There is enough evidence throughout both prior definition and usage sites that support users being expected to test for `NULL`. I think this is a case of outdated/misleading documentation that needs to be updated, rather than backwards-compatibility we need to maintain.

Reply:
I agree the documentation does not match the current behavior. `mongoc_client_pool_new` currently returns `NULL` if [attempting to set TLS options without TLS enabled](](https://github.com/mongodb/mongo-c-driver/blob/a203bafab7d412ed05f33b99229bacbc043c2268/src/libmongoc/src/mongoc/mongoc-client-pool.c#L113-L119)).